src/main/java/cuchaz/enigma/network/packet/UserListS2CPacket.java

```diff
@@ -0,0 +1,51 @@
+package cuchaz.enigma.network.packet;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import cuchaz.enigma.gui.GuiController;
+
+public class UserListS2CPacket implements Packet<GuiController> {
+
+	private List<String> users;
+
+	UserListS2CPacket() {
+	}
+
+	public UserListS2CPacket(List<String> users) {
+		this.users = users;
+	}
+
+	@Override
+	public void read(DataInput input) throws IOException {
+		int len = input.readInt();
+		if (len == 0) {
+			users = Collections.emptyList();
+		} else if (len == 1) {
+			users = Collections.singletonList(input.readUTF());
+		} else {
+			users = new ArrayList<>(len);
+			for (int i = 0; i < len; i++) {
+				users.add(input.readUTF());
+			}
+		}
+	}
+
+	@Override
+	public void write(DataOutput output) throws IOException {
+		output.writeInt(users.size());
```

Comment:
Are we really going to accept 2147483647 users at once? I would add a limit to the numbers allowed on a server at once, as well as reflect that limit in the datatype here. I suggest either `unsigned byte` or `unsigned short`.

Reply:
Ended up changing it to unsigned short.